src/CoreFoundation/CFMutableString.cs

```diff
@@ -0,0 +1,63 @@
+#if !COREBUILD
+
+using System;
+using System.Runtime.InteropServices;
+
+using ObjCRuntime;
+
+namespace CoreFoundation {
+	
+	public class CFMutableString : CFString {
+
+		[DllImport (Constants.CoreFoundationLibrary, CharSet=CharSet.Unicode)]
+		static extern unsafe /* CFMutableStringRef* */ IntPtr CFStringCreateMutableWithExternalCharactersNoCopy (/* CFAllocatorRef* */ IntPtr alloc, string chars, nint numChars, nint capacity, /* CFAllocatorRef* */ IntPtr externalCharactersAllocator);
+
+		public CFMutableString (string theString, nint capacity)
+		{
+			if (theString == null)
+				throw new ArgumentNullException (nameof (theString));
+			
+			handle = CFStringCreateMutableWithExternalCharactersNoCopy (IntPtr.Zero, theString, theString.Length, capacity, IntPtr.Zero);
+		}
+
+		[DllImport (Constants.CoreFoundationLibrary)]
+		static extern unsafe /* CFMutableStringRef* */ IntPtr CFStringCreateMutable (/* CFAllocatorRef* */ IntPtr alloc, nint maxLength);
+
+		public CFMutableString (nint maxLength)
+		{
+			handle = CFStringCreateMutable (IntPtr.Zero, maxLength);
+		}
+
+		[DllImport (Constants.CoreFoundationLibrary)]
+		static extern unsafe /* CFMutableStringRef* */ IntPtr CFStringCreateMutableCopy (/* CFAllocatorRef* */ IntPtr alloc, nint maxLength, /* CFStringRef* */ IntPtr theString);
+
+		public CFMutableString (CFString theString, nint maxLength)
+		{
+			handle = CFStringCreateMutableCopy (IntPtr.Zero, maxLength, theString.GetHandle ());
+		}
+
+		[DllImport (Constants.CoreFoundationLibrary)]
+		[return: MarshalAs (UnmanagedType.I1)]
+		static extern internal bool /* Boolean */ CFStringTransform (/* CFMutableStringRef* */ IntPtr @string, /* CFRange* */ ref CFRange range, /* CFStringRef* */ IntPtr transform, [MarshalAs (UnmanagedType.I1)] /* Boolean */ bool reverse);
+
+		public bool Transform (ref CFRange range, CFStringTransform transform, bool reverse)
+		{
+			str = null; // destroy any cached value
+			return Transform (Handle, ref range, transform.GetConstant ().GetHandle (), reverse);
+		}
+
+		// constant documentation mention it also accept any ICT transform
+		public bool Transform (ref CFRange range, CFString transform, bool reverse)
```

Comment:
What about this API to ease the ICT stuff? `NSString` and `CFString` are toll-free bridged.

```csharp
public bool Transform (ref CFRange range, string transform, bool reverse)
{
	str = null; // destroy any cached value
	var t = NSString.CreateNative (tagClass);
	var ret = CFStringTransform (Handle, ref range, t, reverse);
	NSString.ReleaseNative (t);
	return ret;
}
```

Reply:
I'd need to find real world examples first (for non-constants). It's never 100% clear if the comparison is done on strings (if should) or addresses (should not, but it happens).

Still the next version will be different and use`NSString` because that's how smart enums are exposed :-)

And from here it's not hard to use type`(NSString) String.Empty` if you have a `string` value.

Reply:
done